File: LeetCode_Marathons/LCM6/Medium/Count_Number_of_Homogenous_Substrings/a.go
func countHomogenous(s string) int {
    var module int64 = 1000000007
    var ans int64 = 0
    var keyRune rune = rune('*')
    s += "*"
    var keyCounter int64 = 0
    for _, myCh := range s {
        if myCh == keyRune { 
            keyCounter++ 
        } else {
            var ansForRune int64
            ansForRune = (keyCounter + 1) * keyCounter / 2;
            ans = (ans + ansForRune) % module;
            keyCounter = 1
            keyRune = myCh
        }
    }

    return int(ans);
}

File: LeetCode_Marathons/LCM6/Medium/Count_Number_of_Homogenous_Substrings/a.cpp
class Solution {
public:
    int countHomogenous(string s) {
        long long mod = 1000000007, ans = 0, keyCounter = 0;
        char key = '*';
        s += "*";

        for (char myCh : s) {
            if (myCh == key) keyCounter++;
            else {
                long long ansForKey;
                ansForKey = (keyCounter + 1) * keyCounter / 2;
                ans = (ans + ansForKey) % mod;
                keyCounter = 1;
                key = myCh;
            }
        }

        return ans;
    }
};